nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestValidateRecord.java

```diff
@@ -593,6 +594,47 @@ public void testValidateMaps() throws IOException, InitializationException, Malf
         }
     }
 
+    @Test
+    void testTimeZoneWithTimeStamp() throws Exception {
+        final TimeZone defaultTimezone = TimeZone.getDefault();
+        //Ensure the timezone is UTC in order to validate the use of a timezone in the timestamp pattern is taken into account.
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+        final String timestampWithTimeZonePattern = "EEE, dd MMM yyyy HH:mm:ssZ";
+        final String schema = new String(Files.readAllBytes(Paths.get("src/test/resources/TestValidateRecord/timestampWithTimeZone.avsc")), "UTF-8");
+        final CSVReader csvReader = new CSVReader();
+        runner.addControllerService("reader", csvReader);
+        runner.setProperty(csvReader, SchemaAccessUtils.SCHEMA_ACCESS_STRATEGY, SchemaAccessUtils.SCHEMA_TEXT_PROPERTY);
+        runner.setProperty(csvReader, SchemaAccessUtils.SCHEMA_TEXT, schema);
+        runner.setProperty(csvReader, CSVUtils.FIRST_LINE_IS_HEADER, "true");
+        runner.setProperty(csvReader, CSVUtils.VALUE_SEPARATOR, "◆");
+        runner.setProperty(csvReader, DateTimeUtils.TIMESTAMP_FORMAT, timestampWithTimeZonePattern);
+        runner.enableControllerService(csvReader);
+
+        final CSVRecordSetWriter csvWriter = new CSVRecordSetWriter();
+        runner.addControllerService("writer", csvWriter);
+        runner.setProperty(csvWriter, "Schema Write Strategy", "full-schema-attribute");
+        runner.setProperty(csvWriter, DateTimeUtils.TIMESTAMP_FORMAT, timestampWithTimeZonePattern);
+        runner.enableControllerService(csvWriter);
+
+        runner.setProperty(ValidateRecord.RECORD_READER, "reader");
+        runner.setProperty(ValidateRecord.RECORD_WRITER, "writer");
+        runner.setProperty(ValidateRecord.ALLOW_EXTRA_FIELDS, "false");
+        runner.setProperty(ValidateRecord.MAX_VALIDATION_DETAILS_LENGTH, "4000");
+        runner.setProperty(ValidateRecord.VALIDATION_DETAILS_ATTRIBUTE_NAME, "valDetails");
+
+        final String content = "apache_date◆apache_ip_source◆apache_method◆apache_path◆apache_query_string◆apache_response_code◆apache_referer◆apache_user_agent\n" +
+                "Wed, 24 Jul 2024 15:04:23+0200◆10.4.3.20◆GET◆/path◆?test=toto◆200◆-◆";
+
+        runner.enqueue(content);
+        runner.run();
+
+        runner.assertTransferCount(ValidateRecord.REL_VALID, 1);
+        final MockFlowFile validFlowFile = runner.getFlowFilesForRelationship(ValidateRecord.REL_VALID).get(0);
+        //Validate timezone is taken into account
+        assertTrue(validFlowFile.getContent().contains("Wed, 24 Jul 2024 13:04:23+0000"));
```

Comment:
This value should match the input value to confirm that no unexpected conversion is occurring. Recommend assigning a variable for the date stamp and using it for both input and expected values.

Reply:
Actually this tests that the output is different than the input as the input was 
`Wed, 24 Jul 2024 15:04:23+0200◆10.4.3.20` and the assertion is the timezone was taken into account  and hence an offset in the time to `13:04:23`

Reply:
Thanks for the explanation. I thought that might be the case, however, a goal of the test should be to assert the same values. As a timestamp without a zone is inherently local, the value should not be recalculated, so I recommend designing the test with that in mind.

Reply:
I am a little confused. Shouldn't I be testing when a timezone is specified in the pattern that it is taken into account for the timestamp?

Reply:
Sorry for the confusion. The reported problem is the inclusion of the time zone in the format, but that conflicts with the current conversion approach where `java.sql.Timestamp` does not understand a time zone.

I'm thinking this may need to be addressed at a more fundamental level, but I have not evaluated all the options thus far.

Reply:
If you can get the tests passing on all platforms, that could be helpful as a basis for consideration. However, it may require several iterations, so it might be better to work on that in your fork with GitHub actions first, and then reconsider the pull request at that point. I would like to take a closer look at the issue soon, and I can share additional thoughts in Jira after addressing some other things.

Reply:
@exceptionfactory You were %100 right it took a few iterations to get this right. But after this [search](https://github.com/search?q=repo%3Aapache%2Fnifi++Excel&type=pullrequests&p=2) I remembered I had locale issues when developing the Excel Record Reader in #7194. When I set the locales I realized I could not use this `"Wed, 24 Jul 2024 15:04:23+0200"` so I ended up removing the three character day of the week, the three day month and the `+0200` timezone.  I also tweeked the timezone regular expression to cover all cases.
Please let me know what else needs to be done. Thanks!